### PR TITLE
Intrepid2 - serial interface test and example for high order elements (QUAD Cn, HEX Cn)

### PR DESCRIPTION
**Pull request by kyungjoo-kim**
Intrepid2 serial interface tests for high order elements.

- getWorksizePerPoint should be moved into top-level (we never allocate anything inside of functors; therefore, it is not necessary to get such information from functor level).
- getVandermodeInverse interface is added for high order elements to allow the access.
- test compares result computed by serial interface and top-level interface.
- QUAD and HEX Cn elements are converted. 
- TRI and TET have an issue that I cannot resolve. It looks like that they are not well tested. @mperego take a look at them after you come back.

### CODE REVIEW

**Comment by kyungjoo-kim** [comment]
#2859

File: packages/intrepid2/src/Discretization/Basis/Intrepid2_HGRAD_QUAD_Cn_FEM.hpp [hunk 1]
```diff
@@ -174,15 +170,20 @@ namespace Intrepid2 {
     typedef typename Basis<ExecSpaceType,outputValueType,pointValueType>::ordinal_type_array_2d_host ordinal_type_array_2d_host;
     typedef typename Basis<ExecSpaceType,outputValueType,pointValueType>::ordinal_type_array_3d_host ordinal_type_array_3d_host;
 
+    typedef typename Basis<ExecSpaceType,outputValueType,pointValueType>::outputViewType outputViewType;
+    typedef typename Basis<ExecSpaceType,outputValueType,pointValueType>::pointViewType  pointViewType;
+    typedef typename Basis<ExecSpaceType,outputValueType,pointValueType>::scalarViewType  scalarViewType;
+
+  private:
+    /** \brief inverse of Generalized Vandermonde matrix (isotropic order */
```
Comment (Rombur) [inline packages/intrepid2/src/Discretization/Basis/Intrepid2_HGRAD_QUAD_Cn_FEM.hpp]: You forgot the `)`

File: packages/intrepid2/unit-test/Discretization/Basis/HGRAD_HEX_Cn_FEM/test_02.hpp [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+          
+          // compute setup
+          //   we need vinv and workspace
+          const auto vinv = basis.getVandermondeInverse();
+          
+          // worksize 
+          //   workspace per thread is required for serial interface. 
+          //   parallel_for with range policy would be good to use stack workspace 
+          //   as team policy only can create shared memory 
+          //   this part would be tricky as ths max size should be determined in compile time
```
Comment (Rombur) [inline packages/intrepid2/unit-test/Discretization/Basis/HGRAD_HEX_Cn_FEM/test_02.hpp]: typos: 
  - ths
 - in compile time -> at compile time

File: packages/intrepid2/unit-test/Discretization/Basis/HGRAD_HEX_Cn_FEM/test_02.hpp [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+          //   let's think about this and find out the best practice. for now I use the following.
+          constexpr int worksize = (Parameters::MaxOrder+1)*4;
+          
+          // if you use team policy, worksize can be gathered from the basis object and use 
+          // kokkos shemem_size APIs to create workspace per team or per thread.
```
Comment (Rombur) [inline packages/intrepid2/unit-test/Discretization/Basis/HGRAD_HEX_Cn_FEM/test_02.hpp]: typo: shemem_size -> shmem_size

File: packages/intrepid2/unit-test/Discretization/Basis/HGRAD_TET_Cn_FEM/test_02.hpp [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+          
+          // compute setup
+          //   we need vinv and workspace
+          const auto vinv = basis.getVandermondeInverse();
+          
+          // worksize 
+          //   workspace per thread is required for serial interface. 
+          //   parallel_for with range policy would be good to use stack workspace 
+          //   as team policy only can create shared memory 
+          //   this part would be tricky as ths max size should be determined in compile time
```
Comment (Rombur) [inline packages/intrepid2/unit-test/Discretization/Basis/HGRAD_TET_Cn_FEM/test_02.hpp]: typos: ths and in compile time

File: packages/intrepid2/unit-test/Discretization/Basis/HGRAD_TRI_Cn_FEM/test_02.hpp [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+          
+          // compute setup
+          //   we need vinv and workspace
+          const auto vinv = basis.getVandermondeInverse();
+          
+          // worksize 
+          //   workspace per thread is required for serial interface. 
+          //   parallel_for with range policy would be good to use stack workspace 
+          //   as team policy only can create shared memory 
+          //   this part would be tricky as ths max size should be determined in compile time
```
Comment (Rombur) [inline packages/intrepid2/unit-test/Discretization/Basis/HGRAD_TRI_Cn_FEM/test_02.hpp]: same typos

**Comment by kyungjoo-kim** [comment]
@Rombur Now I see you github ID. I did not put you as a reviewer because I cannot imagine your ID (autocomplete does not find you).

**Comment by kyungjoo-kim** [comment]
@Rombur It may need your approval to merge it as you review the code.

**Pull request merged**